.github/workflows/performance_harness_run.yaml

```diff
@@ -12,10 +22,100 @@ defaults:
     shell: bash
 
 jobs:
-  tmp:
-    name: Stub
+
+  v:
+    name: Discover Inputs
+    runs-on: ubuntu-latest
+    outputs:
+      test-params: ${{steps.overrides.outputs.test-params}}
+    steps:
+      - name: Setup Input Params
+        id: overrides
+        run: |
+          echo test-params=testBpOpMode >> $GITHUB_OUTPUT
+
+          if [[ "${{inputs.override-test-params}}" != "" ]]; then
+            echo test-params=${{inputs.override-test-params}} >> $GITHUB_OUTPUT
+          fi
+
+  platforms:
+    name: Run Platforms Workflow
+    uses: ./.github/workflows/platforms.yaml
+    with:
+      override-build-matrix: ${{github.event.inputs.platform-choice}}
+    permissions:
+      packages: write
+      contents: read
+
+  reuse-build:
+    name: Reuse leap build
+    needs: [v]
+    if: |
+      always() &&
+      needs.v.result == 'success'
+    runs-on: ubuntu-latest
+    outputs:
+      build-artifact: ${{steps.downloadBuild.outputs.downloaded-file}}
+    steps:
+        - name: Download builddir
+          id: downloadBuild
+          uses: AntelopeIO/asset-artifact-download-action@v3
+          with:
+            owner: AntelopeIO
+            repo: leap
+            file: build.tar.zst
+            target: ${{github.sha}}
+            artifact-name: ${{github.event.inputs.platform-choice}}-build
+            fail-on-missing-target: false
+            token: ${{github.token}}
+            
+        - name: Upload builddir
+          if: steps.downloadBuild.outputs.downloaded-file != ''
+          uses: AntelopeIO/upload-artifact-large-chunks-action@v1
+          with:
+            name: ${{github.event.inputs.platform-choice}}-build
+            path: build.tar.zst
+
+  build-base:
+    name: Run Build Workflow
+    needs: [platforms, reuse-build]
+    if: always() && needs.platforms.result == 'success' && needs.reuse-build.outputs.build-artifact == ''
+    uses: ./.github/workflows/build_base.yaml
+    with:
+      p: ${{needs.platforms.outputs.p}}
+      platform-matrix: ${{needs.platforms.outputs.platform-matrix}}
+    permissions:
+      packages: write
+      contents: read
+
+  tests:
+    name: Tests
+    needs: [v, platforms, reuse-build, build-base]
+    if: always() && needs.platforms.result == 'success' && (needs.build-base.result == 'success' || needs.reuse-build.result == 'success')
```

Comment:
we really need to find a way to get #283 in good shape to avoid these tricky `always()` chains